Generate docs for dotnet nuget commands. (#3253)
NuGet/NuGet.Client: src/NuGet.Core/NuGet.CommandLine.XPlat/external/docs.md

    

---
title: dotnet nuget add source command

> description: The `dotnet nuget add source` command will add a new source to your NuGet configuration files. This will enable finding packages at this additional source. 

Reviewer: What do you think of this?

```suggestion
description: The `dotnet nuget add source` command adds a new package source to your NuGet configuration files.
```

The two sentences are a little similar, hence why I merged them.

Author: yep. made the change.

author: nugetClient
ms.date: 2/21/2020
---
# dotnet nuget add source

**This article applies to:** ✔️ .NET Core 3.1.200 SDK and later versions

## Name

`dotnet nuget add source` - Add a NuGet source.

## Synopsis

```dotnetcli
`dotnet nuget add source PACKAGESOURCEPATH [--name] [--username] [--password] [--store-password-in-clear-text] [--valid-authentication-types] [--configfile]`
`dotnet nuget add source [-h|--help]`
```

## Description

The `dotnet nuget add source` command will add a new source to your NuGet configuration files. This will enable finding packages at this additional source. 

## Arguments
- **`PACKAGESOURCEPATH`**

  Path to the package(s) source.

## Options
- **`-n|--name`**

  Name of the source.

- **`-u|--username`**

  UserName to be used when connecting to an authenticated source.

- **`-p|--password`**

  Password to be used when connecting to an authenticated source.

> - **`--store-Password-In-Clear-Text`**

Reviewer: Should this be lowercase to match the CLI option?

Reviewer: Same comment for `valid-Authentication-Types`

Author: yep...gonna send another commit. missed this comment.

  Enables storing portable package source credentials by disabling password encryption.

- **`--valid-Authentication-Types`**

>   Comma-separated list of valid authentication types for this source. By default, all authentication types are valid. Example: basic,negotiate

Reviewer: What are all the available authentication types? Should we give a quick synopsis on what `basic` and `negotiate` means? Are there additional docs on these types that we can link? I would recommend wrapping the example in a code block: `basic,negotiate`

Author: will open doc issue to improve the details here...not blocking now.
tracking here: https://github.com/NuGet/Home/issues/9233

  Path to the package(s) source.

- **`-u|--username`**

  UserName to be used when connecting to an authenticated source.

- **`-p|--password`**

  Password to be used when connecting to an authenticated source.

- **`--store-Password-In-Clear-Text`**

  Enables storing portable package source credentials by disabling password encryption.

- **`--valid-Authentication-Types`**

  Comma-separated list of valid authentication types for this source. By default, all authentication types are valid. Example: basic,negotiate

- **`--configfile`**

  The NuGet configuration file. If specified, only the settings from this file will be used. If not specified, the hierarchy of configuration files from the current directory will be used. To learn more about NuGet configuration go to https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior.

## Examples

- Enable a source with name of `mySource`:

  ```dotnetcli
  dotnet nuget remove source mySource
  ```
## See also

- [Package source sections in NuGet.config files](/nuget/reference/nuget-config-file#package-source-sections)

- [sources command (nuget.exe)](/nuget/reference/cli-reference/cli-ref-sources)

> }

Reviewer: I'm not familiar with this file's syntax - is this `}` intentional?

Author: nope. thx. fixing

    
---file:/dotnet/core/tools/dotnet-nuget-add-source
---
title: dotnet nuget add source command
description: The `dotnet nuget add source` command adds a new package source to your NuGet configuration files. 
author: nugetClient
ms.date: 03/02/2020
---
# dotnet nuget add source

**This article applies to:** ✔️ .NET Core 3.1.200 SDK and later versions

## Name

`dotnet nuget add source` - Add a NuGet source.

## Synopsis

```dotnetcli
`dotnet nuget add source PACKAGESOURCEPATH [--name] [--username] [--password] [--store-password-in-clear-text] [--valid-authentication-types] [--configfile]`
`dotnet nuget add source [-h|--help]`
```

## Description

The `dotnet nuget add source` command adds a new package source to your NuGet configuration files. 

## Arguments

- **`PACKAGESOURCEPATH`**

  Path to the package(s) source.

## Options

- **`-n|--name`**

  Name of the source.

- **`-u|--username`**

>   UserName to be used when connecting to an authenticated source.

Reviewer: Should this be `Username`? It’s a valid English word: https://www.merriam-webster.com/dictionary/username

Reviewer: There’s several occurrences of `UserName`, don’t forget to update those too if you change this.

Author: fixed in next commit. also fixed strings.resx (nuget.exe command)

## Examples

- Enable a source with name of `mySource`:

  ```dotnetcli
  dotnet nuget enable source mySource
  ```

## See also

- [Package source sections in NuGet.config files](/nuget/reference/nuget-config-file#package-source-sections)

- [sources command (nuget.exe)](/nuget/reference/cli-reference/cli-ref-sources)
    
---file:/dotnet/core/tools/dotnet-nuget-list-source
---
title: dotnet nuget list source command
description: The `dotnet nuget list source` command lists all existing sources from your NuGet configuration files. 
author: nugetClient
ms.date: 03/02/2020
---
# dotnet nuget list source

**This article applies to:** ✔️ .NET Core 3.1.200 SDK and later versions

## Name

`dotnet nuget list source` - Lists all configured NuGet sources.

## Synopsis

```dotnetcli
`dotnet nuget list source [--format] [--configfile]`
`dotnet nuget list source [-h|--help]`
```

## Description

The `dotnet nuget list source` command lists all existing sources from your NuGet configuration files. 

## Options

- **`--format`**

>   Applies to the list action. Accepts two values: Detailed (the default) and Short.

Reviewer: What do you think of something like `The format of the list command output` to clarify this affects the output? Also can we wrap `Detailed` and `Short` in back ticks?

Author: fixed in next commit

`dotnet nuget update source [-h|--help]`
```

## Description

The `dotnet nuget update source` command updates an existing source in your NuGet configuration files. 

## Arguments

- **`NAME`**

  Name of the source.

## Options

- **`-s|--source`**

  Path to the package(s) source.

- **`-u|--username`**

  UserName to be used when connecting to an authenticated source.

- **`-p|--password`**

  Password to be used when connecting to an authenticated source.

- **`--store-password-in-clear-text`**

  Enables storing portable package source credentials by disabling password encryption.

- **`--valid-authentication-types`**

  Comma-separated list of valid authentication types for this source. By default, all authentication types are valid. Example: basic,negotiate

- **`--configfile`**

  The NuGet configuration file. If specified, only the settings from this file will be used. If not specified, the hierarchy of configuration files from the current directory will be used. To learn more about NuGet configuration go to https://docs.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior.

## Examples

- Enable a source with name of `mySource`:

  ```dotnetcli

>   dotnet nuget remove source mySource

Reviewer: Is this the right example? Shouldn’t this be an update?

Author: fixed in next commit
